### DISCUSSION TITLE
Android MPD writeable playlist directory

### DISCUSSION THREAD

**Discussion by ckrosco** [MusicPlayerDaemon/MPD#1229] [Q&A]
Where can I have writeable playlists on an android TV box?

I've installed MPD on an android tv box and it is working great, except I cannot save playlists.

I am editing the mpd.conf file by ssh. Busybox is installed.

I've tried the following without success:

playlist_directory  "/sdcard/Music/playlists"
playlist_directory  "/data/data/org.musicpd/cache/playlists"
playlist_directory  "/storage/emulated/0/Music/playlists"
playlist_directory  "/mnt/sdcard/Music/playlists"
playlist_directory  "/data/data/org.musicpd/files"
playlist_directory  "/data/playlists"

I changed the permissions on these directories to 777 using "su", but when checking permissions and ownerships with the "user" account, owner and (sometimes) permissions are different from what I see using the "su" (root) account.

I am unfamiliar with android - any help is appreciated.

**Comment by MaxKellermann** [comment]
What does "without success" mean, exactly?

**Reply by ckrosco** [reply]
I am not able to save playlists to any of those directories.

My music files are accessed by nfs -- music_directory "nfs://192.168.1.9/export/CDs"
This works great - any client can list and play all my music files. But when I have a queue and try to save it (e.g. using MPDroid), I am asked to name the new playlist and save. MPDroid does not report any errors, and the playlist is not saved. 

This message shows up in MPD on the android tv:
D/ client: [11] process command "save "test""
E/ exception: Failed to create /data/playlists/test.m3u: Permission denied

So then I modified the mpd.conf file without a playlist_directory thinking there might be a default location:
D/ client: [27] process command "listplaylistinfo "playlist1""
E/ exception: Failed to open '/sdcard/Music/playlists/playlist1.m3u': Permission denied
E/ exception: NFS: Lookup of /playlist1 failed with NFS3ERR_NOENT(-2)

Finally I tried using an nfs path for the playlist:
E/ exception: Error on line 8: not an absolute path: nfs://192.168.1.9/export/CDs/playlists

So my question is: Where can I save playlists?

**Reply by MaxKellermann** [reply]
Which Android version is that?
Newer Android versions have very strict permissions, and don't allow apps to write anything anywhere, except for one app-specific directory. That makes the lives of app developers very difficult, but I guess we should hard-code the playlist directory on Android to that one writable directory.
Please open an issue for that, and I'll check it out.

**Reply by ckrosco** [reply]
Android 6.0.1

I'll open an issue. This is a fabulous piece of software and I appreciate all your work on it.
